install/includes/functions.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+/**
+ * Checks if required step is in session, if not redirect to $failureStep
+ *
+ * @param integer $requiredStep
+ * @param integer $failureStep
+ * @param bool $equal
+ */
+function checkStep($requiredStep, $failureStep, $equal = true)
+{
+    if (!array_key_exists('step', $_SESSION) || ($equal === true && $_SESSION['step'] !== $requiredStep) || ($equal === false && $_SESSION['step'] < $requiredStep)) {
+        // Redirect to failure-step
+        header('location: ' . $_SERVER['PHP_SELF'] . '?step=' . $failureStep);
+        exit;
+    }
+}
+
+/**
+ * Configures the PHP-Extension "gettext" to use the configured locale
+ *
+ * @param $locale
+ * @param $defaultLocale
+ */
+function configureGettextExtension($locale, $defaultLocale)
+{
+    // Check for module "gettext"
+    if (!function_exists('gettext')) {
+        // @ToDo: real error-handling
+        die('<strong>Error:</strong> The PHP-Extension "gettext" is missing so it is not possible to change the language!');
+    }
+
+    // Set translation-domain
+    bindtextdomain('poweradmin_installer', realpath(INSTALLER_DIRECTORY . 'locales'));
+    bind_textdomain_codeset('poweradmin_installer', 'UTF-8');
+    textdomain('poweradmin_installer');
+
+    // Set locale
+    $result = setlocale(LC_ALL, $locale);
+    putenv('LANG=' . $locale);
+    putenv('LANGUAGE=' . $locale);
+
+    // Check for error -> locale not exist
+    if (!$result && $locale !== $defaultLocale) {
+        // Try to set default locale
+        $result = setlocale(LC_ALL, $defaultLocale);
+        putenv('LANG=' . $defaultLocale);
+        putenv('LANGUAGE=' . $defaultLocale);
+    }
+
+    // Check for error -> locale and even default locale not exist
+    if (!$result) {
```

Comment:
It simply fails with message: Error: The locale "en_US" does not exist!
And this will happen to other user as well. Another issue that I found with this English locale is:
new installer creates config with en_US, but there's en_EN directory in locale directory, also
selection of default English language fails en_US != en_EN.